### ISSUE TITLE
Unable to disable endpoint authentication

### ISSUE THREAD

**Issue by ryanemerson** [infinispan/infinispan-images#88]
Due to the `security-realm` being configured on the `<endpoints ..>` element, authentication is always enabled for endpoints regardless of whether the `<authentication..>` element is added. We should update the generated xml to the following:

```xml
      <endpoints socket-binding="default" >
         <!-- Add security-realm attribute and authentication element only if auth is enabled for each endpoint -->
         <hotrod-connector name="hotrod" security-realm="default">
            <authentication>
              <sasl mechanisms="SCRAM-SHA-512 SCRAM-SHA-384 SCRAM-SHA-256 SCRAM-SHA-1 DIGEST-SHA-512 DIGEST-SHA-384 DIGEST-SHA-256 DIGEST-SHA DIGEST-MD5 PLAIN"
                          qop="auth" server-name="infinispan"/>
              </authentication>
         </hotrod-connector>
         <rest-connector name="rest"/>
      </endpoints>
``

**Comment by ryanemerson** [comment]
Requires a fix for [ISPN-12320](https://issues.redhat.com/browse/ISPN-12320).

**Comment by ryanemerson** [comment]
Fixed from 12.0.x

**Comment by ryanemerson** [comment]
Actually fixed in config-generator 2.2.Final. It's not possible to disable authentication per endpoint as this does not make sense when exposing the endpoints on a single port. It's only possible to disable auth for all endpoints via `endpoints.auth: false`.
